### PR TITLE
refactor: Make proxy handlers into singletons

### PR DESCRIPTION
**Pull request by dmaretskyi**
### Motivation / Background

This Pull Request ...

### CODE REVIEW

File: packages/core/echo/echo-schema/src/effect/echo-handler.ts [hunk 1]
```diff
@@ -405,44 +496,48 @@ export class EchoReactiveHandlerImpl extends EchoReactiveHandler implements Reac
...
-    return target;
+    return target as EchoArrayTwoPointO<any>;
   }
 
-  getMeta(): ObjectMeta {
-    const target: any = { [symbolPath]: [], [symbolNamespace]: META_NAMESPACE };
-    return createReactiveProxy(target, this) as ObjectMeta;
+  getMeta(target: ProxyTarget): ObjectMeta {
+    const metaTarget: ProxyTarget = {
+      [symbolInternals]: target[symbolInternals],
```
Comment (yarolegovich) [inline packages/core/echo/echo-schema/src/effect/echo-handler.ts:527]: dataPath doesn't include namespace, so should be a separate map for meta proxy or full path should be used when doing lookup

File: packages/core/echo/echo-schema/src/effect/echo-handler.ts [hunk 2]
```diff
@@ -573,42 +630,60 @@ export const createEchoReactiveObject = <T extends {}>(init: T): EchoReactiveObj
...
-    const echoHandler = new EchoReactiveHandlerImpl();
-    echoHandler._objectCore.rootProxy = proxy;
+    const core = new AutomergeObjectCore();
+    core.rootProxy = proxy;
 
-    slot.handler = echoHandler;
+    slot.handler = EchoReactiveHandlerImpl.instance;
     const target = slot.target as ProxyTarget;
+
+    target[symbolInternals] = {
```
Comment (yarolegovich) [inline packages/core/echo/echo-schema/src/effect/echo-handler.ts:639]: nit: extract into createObjectInternals function

File: packages/core/echo/echo-schema/src/effect/echo-handler.ts (hunk 1 above)
Comment (dmaretskyi) [reply packages/core/echo/echo-schema/src/effect/echo-handler.ts:527]: for targets map?

yeah, its also broken for expandos - see my comment on where it is declared

Comment (yarolegovich) [reply packages/core/echo/echo-schema/src/effect/echo-handler.ts:527]: I mean imagine you have `struct({ keys: S.array(S.any) })`
here you will create an object and cache it using `['keys']` key
https://github.com/dxos/dxos/blob/2ec80d874299475b4718b958e7447f503c9ec3d2/packages/core/echo/echo-schema/src/effect/echo-handler.ts#L249
but then you request a meta object and when someone accesses keys they get the cache target from `data` namespace

Comment (dmaretskyi) [reply packages/core/echo/echo-schema/src/effect/echo-handler.ts:562]: done

**Pull request merged**